mono/mini/interp/interp.c

```diff
@@ -2907,13 +2907,29 @@ static int opcode_counts[512];
 #define frame_objref(frame) o
 #endif
 
+// Leave is split into pieces in order to consume less stack,
+// but not have to change how exception handling macros access labels and locals.
+static MONO_NEVER_INLINE MonoException*
+mono_interp_leave (InterpFrame* child_frame)
```

Comment:
Maybe name this mono_interp_get_undeniable_exception or something like that. The current name is not really representative of what this method is doing.

Reply:
To naïve/shallow/empirical me, it is `mono_interp_outlined_mechanical_part_of_leave_whatever_it_takes`, shortened to `mono_interp_leave`, not anything more general or more specific, not be used anywhere else. But of course I try to be flexible. :)

Reply:
`mono_interp_leave_outlined`??
There might be another chunk, I was going to call them `mono_interp_leave1` and `mono_interp_leave2`.
I need to double check that it useful for purposes of stack.

Reply:
I see what you mean though, it is one line doing just that.
Nitty question, then..
```					child_frame.imethod = NULL; 
					child_frame.imethod = NULL;
```
should be moved into this function, for it to be more complete/sensible per your name?
For stack preservation they can go in either place. I only moved the minimal amount, didn't attempt to make a sensible function.